ThunkLibs/libX11/libX11_Host.cpp

```diff
@@ -57,54 +50,13 @@ char* fexfn_impl_libX11_XGetICValues_internal(XIC a_0, size_t count, unsigned lo
     }
 }
 
-struct XIfEventCB_args {
-    XIfEventCBFN *fn;
-    XPointer arg;
-};
-
-static int XIfEventCB(Display* a0, XEvent* a1, XPointer a2) {
-
-    XIfEventCB_args *arg = (XIfEventCB_args*)a2;
-
-    XIfEventCB_Args argsrv { a0, a1, arg->arg};
-
-    call_guest(callback_unpacks->libX11_XIfEventCB, (void*) arg->fn, &argsrv);
-    
-    return argsrv.rv;
-}
-
-int fexfn_impl_libX11_XIfEvent_internal(Display* a0, XEvent* a1, XIfEventCBFN* a2, XPointer a3) {
-    static XIfEventCB_args args = { a2, a3 };
+XErrorHandler guest_handler {};
 
-    return fexldr_ptr_libX11_XIfEvent(a0, a1, &XIfEventCB, (XPointer)&args);
-}
-
-Bool fexfn_impl_libX11_XUnregisterIMInstantiateCallback_internal(
-    Display*, struct _XrmHashBucketRec*,
-    char*, char*, XUnregisterIMInstantiateCallbackCBFN*, XPointer) {
-    fprintf(stderr, "XUnregisterIMInstantiateCallback: Stubbed");
-    return true;
-}
-
-void fexfn_impl_libX11_XRemoveConnectionWatch_internal(Display*, XRemoveConnectionWatchCBFN*, XPointer) {
-    fprintf(stderr, "XRemoveConnectionWatch: Stubbed");
-}
-
-XErrorHandler guest_handler;
-
-int XSetErrorHandlerCB(Display* a_0, XErrorEvent* a_1) {
-    static XSetErrorHandlerCB_Args argsrv { a_0, a_1};
-    
-    call_guest(callback_unpacks->libX11_XSetErrorHandlerCB, (void*) guest_handler, &argsrv);
-    
-    return argsrv.rv;
-}
-
-XSetErrorHandlerCBFN* fexfn_impl_libX11_XSetErrorHandler_internal(XErrorHandler a_0) {
+XErrorHandler fexfn_impl_libX11_XSetErrorHandler(XErrorHandler a_0) {
```

Comment:
as `a_0` is no longer a guest handler, but a host handler, we can just return directly instead of storing in `XErrorHandler guest_handler`.